pkg/controller/controller.go

```diff
@@ -185,8 +189,8 @@ func worker(queue workqueue.RateLimitingInterface, resourceType string, maxRetri
 				}
 				defer queue.Done(key)
 
-				err := reconciler(key.(string))
-				if err == nil {
+				completed, err := reconciler(key.(string))
+				if completed {
 					if forgetAfterSuccess {
 						queue.Forget(key)
 					}
```

Comment:
What happens if "err" isn't nil? Do we log it someplace or just drop it?  Even if we do log it as part of the normal processing of the "reconciler", it would be nice to add something to the log to indicate that despite the error we're stopping anyway and will not try again. Will make it easier to debug.

Reply:
The errors are, for the most part, logged already in numerous ways in the reconcile functions themselves. I would not be opposed to adding a low-level info log here. Any higher severity logging I think would be redundant.

Reply:
To be clear, if we're already logging it I'm not suggesting we log the same error message again, I'm looking for us to log something that says "we're giving up" so people know that this time around is different from the rest.

Reply:
I do think that we should log / put a status message on a resource when we give up.

Reply:
+1 to having a log message should the controller receive an error but decides to not add back to the queue.